src/plugins/dashboard/server/content_management/v3/cm_services.ts

```diff
@@ -0,0 +1,540 @@
+        controlConfig: schema.maybe(schema.recordOf(schema.string(), schema.any())),
+        id: schema.string({
+          defaultValue: uuidv4(),
+          meta: { description: 'The unique ID of the control.' },
+        }),
+        order: schema.number({
+          meta: {
+            description: 'The order of the control panel in the control group.',
+          },
+        }),
+        width: schema.oneOf(
+          Object.values(CONTROL_WIDTH_OPTIONS).map((value) => schema.literal(value)) as [
+            Type<ControlWidth>
+          ],
+          {
+            defaultValue: DEFAULT_CONTROL_WIDTH,
+            meta: { description: 'Minimum width of the control panel in the control group.' },
+          }
+        ),
+        grow: schema.boolean({
+          defaultValue: DEFAULT_CONTROL_GROW,
+          meta: { description: 'Expand width of the control panel to fit available space.' },
+        }),
+      },
+      { unknowns: 'allow' }
+    ),
+    {
+      defaultValue: [],
+      meta: { description: 'An array of control panels and their state in the control group.' },
+    }
+  ),
+  labelPosition: schema.oneOf(
+    Object.values(CONTROL_LABEL_POSITION_OPTIONS).map((value) => schema.literal(value)) as [
+      Type<ControlLabelPosition>
+    ],
+    {
+      defaultValue: DEFAULT_CONTROL_LABEL_POSITION,
+      meta: {
+        description: 'Position of the labels for controls. For example, "oneLine", "twoLine".',
+      },
+    }
+  ),
+  chainingSystem: schema.oneOf(
+    Object.values(CONTROL_CHAINING_OPTIONS).map((value) => schema.literal(value)) as [
+      Type<ControlGroupChainingSystem>
+    ],
+    {
+      defaultValue: DEFAULT_CONTROL_CHAINING,
+      meta: {
+        description:
+          'The chaining strategy for multiple controls. For example, "HIERARCHICAL" or "NONE".',
+      },
+    }
+  ),
+  enhancements: schema.maybe(schema.recordOf(schema.string(), schema.any())),
+  ignoreParentSettings: schema.object({
+    ignoreFilters: schema.boolean({
+      meta: { description: 'Ignore global filters in controls.' },
+      defaultValue: DEFAULT_IGNORE_PARENT_SETTINGS.ignoreFilters,
+    }),
+    ignoreQuery: schema.boolean({
+      meta: { description: 'Ignore the global query bar in controls.' },
+      defaultValue: DEFAULT_IGNORE_PARENT_SETTINGS.ignoreQuery,
+    }),
+    ignoreTimerange: schema.boolean({
+      meta: { description: 'Ignore the global time range in controls.' },
+      defaultValue: DEFAULT_IGNORE_PARENT_SETTINGS.ignoreTimerange,
+    }),
+    ignoreValidations: schema.boolean({
+      meta: { description: 'Ignore validations in controls.' },
+      defaultValue: DEFAULT_IGNORE_PARENT_SETTINGS.ignoreValidations,
+    }),
+  }),
+  autoApplySelections: schema.boolean({
+    meta: { description: 'Show apply selections button in controls.' },
+    defaultValue: DEFAULT_AUTO_APPLY_SELECTIONS,
+  }),
+});
+
+const searchSourceSchema = schema.object(
```

Comment:
Should this be defined in data plugin or es-query package and imported here?

Reply:
Eventually I think, yes. The corresponding TypeScript type is [`SerializedSearchSourceFields`](https://github.com/elastic/kibana/blob/main/src/plugins/data/common/search/search_source/types.ts#L124-L169). However, the schema I've implemented here is a subset of that type specific to Dashboards. For example, Dashboards do not support creating filters from ES|QL (typed as [`AggregateQuery`](https://github.com/elastic/kibana/blob/main/src/plugins/data/common/search/search_source/types.ts#L129)). If/when ES|QL filters are added, we can add